5.0/en/0x18-V10-Coding.md

```diff
@@ -78,6 +78,20 @@ Dependency management is critical to the safe operation of any application of an
 | **10.6.1** | [MODIFIED, MOVED FROM 14.2.1] Verify that all components are up to date. | ✓ | ✓ | ✓ | |
 | **10.6.2** | [MODIFIED, MOVED FROM 10.3.2] Verify that third-party components and all of their transitive dependencies are included from the expected repository, whether internally owned or an external source, and that there is no risk of a dependency confusion attack. | ✓ | ✓ | ✓ | 427 |
 
+## V10.7 Concurrency
+
+Without proper synchronization, concurrent access to shared resources can result in corrupted data, system crashes, or unreliable application behavior. Furthermore, race conditions can often be chained to perform privilege escalations or remote code execution.
+
+| # | Description | L1 | L2 | L3 | CWE |
+| :---: | :--- | :---: | :---:| :---: | :---: |
+| **10.7.1** | [ADDED, SPLIT FROM 11.1.6, 1.11.2, 1.11.3] Verify that only thread-safe types are used in multi-threaded contexts, or that non-thread-safe types are properly synchronized to prevent race conditions. | | | ✓ | 362 |
+| **10.7.2** | [ADDED, SPLIT FROM 11.1.6, 1.11.2, 1.11.3] Verify that concurrent access to shared resources is controlled using synchronization primitives (e.g., locks, mutexes, semaphores) to prevent race conditions and ensure atomic operations on these resources. | | | ✓ | 366 |
+| **10.7.3** | [ADDED, SPLIT FROM 11.1.6, 1.11.2, 1.11.3] Verify that all access to shared resources is consistently checked and accessed in a single atomic operation to prevent Time-of-Check to Time-of-Use (TOC/TOU) race conditions, ensuring resource state consistency between check and use. | | | ✓ | 367 |
+| **10.7.4** | Verify that resource acquisition follows a consistent lock ordering to prevent circular dependencies that could lead to deadlocks. | | | ✓ | 833 |
+| **10.7.5** | Verify that objects and threads do not cause a livelock scenario by attempting to acquire the same resource and indefinitely yielding to each other without making progress. | | | ✓ | |
```

Comment:
Do you think that 10.7.4 and 10.7.5 could be merged? Are they very different?

Reply:
They're different, but I'm down to merge since the remediating action is fairly similar